src/test/java/com/spotify/github/v3/clients/RepositoryClientTest.java

```diff
@@ -537,4 +539,63 @@ public void mergeNoop() {
     final Optional<CommitItem> maybeCommit = repoClient.merge("basebranch", "headbranch").join();
     assertThat(maybeCommit, is(Optional.empty()));
   }
+
+  @Test
+  public void shouldDownloadTarball() throws Exception {
+    CompletableFuture<Response> fixture = completedFuture(
+        new Response.Builder()
+            .request(new Request.Builder().url("https://example.com/whatever").build())
+            .protocol(Protocol.HTTP_1_1)
+            .message("")
+            .code(200)
+            .body(
+                ResponseBody.create(
+                    "some bytes".getBytes(StandardCharsets.UTF_8),
+                    MediaType.get("application/gzip")
+                ))
+            .build());
+    when(github.request("/repos/someowner/somerepo/tarball/")).thenReturn(fixture);
+
+    try(InputStream response = repoClient.downloadTarball().get().orElseThrow()) {
+      String result = new String(response.readAllBytes(), StandardCharsets.UTF_8);
+      assertThat(result, is("some bytes"));
+    }
+  }
+
+  @Test
+  public void shouldDownloadZipball() throws Exception {
+    CompletableFuture<Response> fixture = completedFuture(
+        new Response.Builder()
+            .request(new Request.Builder().url("https://example.com/whatever").build())
+            .protocol(Protocol.HTTP_1_1)
+            .message("")
+            .code(200)
+            .body(
+                ResponseBody.create(
+                    "some bytes".getBytes(StandardCharsets.UTF_8),
+                    MediaType.get("application/gzip")
+                ))
+            .build());
+    when(github.request("/repos/someowner/somerepo/zipball/")).thenReturn(fixture);
+
+    try (InputStream response = repoClient.downloadZipball().get().orElseThrow()) {
+      String result = new String(response.readAllBytes(), StandardCharsets.UTF_8);
+      assertThat(result, is("some bytes"));
+    }
+  }
+
+  @Test
+  public void shouldReturnEmptyOptionalWhenResponseBodyNotPresent() throws Exception {
+    CompletableFuture<Response> fixture = completedFuture(
+        new Response.Builder()
+            .request(new Request.Builder().url("https://example.com/whatever").build())
+            .protocol(Protocol.HTTP_1_1)
+            .message("")
+            .code(204) // No Content
```

Comment:
Nice that you described the status code with a short comment. :+1: (No need to change anything)